src/mnode/mnode-pastel.cpp

```diff
@@ -1174,13 +1174,68 @@ CArtTradeTicket CArtTradeTicket::Create(std::string _sellTnxId, std::string _buy
     ticket.price = sellTicket->askedPrice;
 
     ticket.GenerateTimestamp();
-    
+
+    // In case it is nested it means that we have the artTnxId of the sell ticket
+    // available within the trade tickets.
+    // [0]: original registration ticket's txid
+    // [1]: copy number for a given art
+    std::vector<std::string> artRegTicket_TxId_Serial = CArtTradeTicket::GetArtRegTxIDAndSerialIfResoldNft(sellTicket->artTnxId);
+    if(artRegTicket_TxId_Serial[0].compare("") == 0)
+    {  
+      auto artTicket = ticket.FindArtRegTicket();
+      if (!artTicket)
+      {
+        throw std::runtime_error("Art Reg ticket not found");
+      }
+      else
+      {
+        //Original TxId
+        ticket.SetArtRegTicketTxid(artTicket->GetTxId());
+        //Copy nr.
+        std::stringstream copy_nr;
+        copy_nr << sellTicket->copyNumber;
+        ticket.SetCopySerialNr(copy_nr.str());
+      }
+    }
+    else
```

Comment:
any case checking that list size is 2?

Reply:
It cannot be else if you are referring to this variable: artRegTicket_TxId_Serial -> It is always 2 and meant to be like this.